tensorflow/compiler/xla/BUILD

```diff
@@ -502,6 +502,7 @@ cc_library(
         ":util",
         ":xla_data_proto_cc",
         "//tensorflow/core:lib",
+        "//tensorflow/core/util/tensor_bundle",
```

Comment:
I noticed that this dependency drags in all of tensorflow. For just a byte_swap that is a fairly high price.

Maybe we can move the basic byte_swap functionality under tensorflow/platform, so that we have a smaller dependency?

Reply:
@sherhut  Yeah, that's true.

I tried to separate the byte_swap functionality from tensor_bundle, but I noticed some other filegroup in tensor_bundle has referenced it as well: 
https://github.com/tensorflow/tensorflow/blob/a9f3b6fd436d84dae9dcdce45d255ba96d6521ba/tensorflow/core/util/tensor_bundle/BUILD#L24-L35

So I think it would be better to add a separate cc_library entry `byteswap` in tensor_bundle to decrease the dependency complexity in other BUILD files. Please take a look at the updated code changes and let me know if you think it's ok. Thank you very much!